src/arrayMethodJoin.js

```diff
@@ -3,9 +3,24 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        str += '';
+      } else {
+        str += this[i];
+      }
```

Comment:
if the element is undefined or null, you do nothing. So you can delete else

```suggestion
      if (typeof this[i] !== 'undefined' && this[i] !== null) {
        str += this[i];
      }
```